File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.java
class Solution {
    public int maximumCount(int[] nums) {
        int low,high,mid,pos,neg;
        low =0;high=nums.length-1;pos=neg=0;
        //positive side
        while(low<=high){
            mid = (low+high)/2;
            if(nums[mid]>0){
                pos += high -mid +1;
                high = mid -1;
            }
            else {
                low = mid +1;
            }
        }
        //negative side
        low =0;high = nums.length-1;
        while(low<=high){
            mid = (low+high)/2;
            if(nums[mid]<0){
                neg += mid -low+1;
                low = mid+1;
            }
            else {
                high = mid -1;
            }
        }
        return pos>neg?pos:neg;
    }

}

File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.go
func maximumCount(nums []int) int {
    var low,high,mid,pos,neg int
    low=0
    high = len(nums)-1
    pos =0
    neg = 0
    for low<=high {
        mid = (low+high)/2
        if nums[mid]>0 {
            pos += high-mid+1
            high = mid -1
        } else {
            low = mid +1
        }
    }
    low =0
    high = len(nums)-1
    for low<=high {
        mid = (low+high)/2
        if nums[mid]<0 {
            neg += mid -low +1
            low = mid+1
        } else {
            high = mid -1
        }
    }
    if pos>neg {
        return pos
    } else {
        return neg
    }
}